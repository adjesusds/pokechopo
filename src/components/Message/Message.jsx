// External Packages
import { node, oneOf } from 'prop-types'
import React from 'react'

// Internal Packages
import StyledMessage from './styled/Message'

/**
 * Creates a new Message component.
 * @class
 */
const Message = ({ children, variant }) => <StyledMessage variant={variant}>{children}</StyledMessage>

/** @override */
Message.propTypes = {
  children: node.isRequired,
  variant: oneOf(['danger', 'info', 'normal', 'success', 'warning']),
}

/** @override */
Message.defaultProps = {
  variant: 'normal',
}

export default Message
