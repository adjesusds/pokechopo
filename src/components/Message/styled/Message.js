// External Packages
import styled from 'styled-components'
import { variants } from 'styled-theming'

// Internal Packages
import { colors, dimensions } from 'config/palette'

/**
 * @constant {Function} * The variants to apply.
 */
const color = variants('mode', 'variant', {
  danger: { normal: colors.red.x500 },
  info: { normal: colors.grey.x500 },
  success: { normal: colors.green.x500 },
  warning: { normal: colors.yellow.x500 },
})

/**
 * @constant {Element} style The style of the component.
 */
const style = styled.span`
  color: ${color};
  font-size: ${dimensions.xLarge};
`

export default style
