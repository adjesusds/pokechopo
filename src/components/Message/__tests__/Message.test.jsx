// External Packages
import { shallow } from 'enzyme'
import React from 'react'

// Internal Packages
import Message from '../Message'

describe('<Message />', () => {
  it('should render the component on valid props', () => {
    let component

    component = shallow(<Message>Hello, World!</Message>)

    expect(component).toMatchSnapshot()

    component = shallow(<Message variant="danger">Hello, World!</Message>)

    expect(component).toMatchSnapshot()

    component = shallow(<Message variant="info">Hello, World!</Message>)

    expect(component).toMatchSnapshot()

    component = shallow(<Message variant="success">Hello, World!</Message>)

    expect(component).toMatchSnapshot()

    component = shallow(<Message variant="warning">Hello, World!</Message>)

    expect(component).toMatchSnapshot()
  })
})
