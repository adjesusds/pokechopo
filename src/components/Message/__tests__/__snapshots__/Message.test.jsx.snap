// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render the component on valid props 1`] = `
<styled.span
  variant="normal"
>
  Hello, World!
</styled.span>
`;

exports[`<Message /> should render the component on valid props 2`] = `
<styled.span
  variant="danger"
>
  Hello, World!
</styled.span>
`;

exports[`<Message /> should render the component on valid props 3`] = `
<styled.span
  variant="info"
>
  Hello, World!
</styled.span>
`;

exports[`<Message /> should render the component on valid props 4`] = `
<styled.span
  variant="success"
>
  Hello, World!
</styled.span>
`;

exports[`<Message /> should render the component on valid props 5`] = `
<styled.span
  variant="warning"
>
  Hello, World!
</styled.span>
`;
